flocker/control/_model.py

```diff
@@ -689,6 +730,57 @@ def __new__(cls, **kwargs):
     def update_cluster_state(self, cluster_state):
         return cluster_state.update_node(self)
 
+    # Attributes that may be set to None to indicate ignorance:
+    _POTENTIALLY_IGNORANT_ATTRIBUTES = ["used_ports", "applications",
+                                        "manifestations", "paths",
+                                        "devices"]
+
+    def _completely_ignorant(self):
+        """
+        Return whether the node is completely ignorant of anything.
+        """
+        return all(getattr(self, attr) is None
+                   for attr in self._POTENTIALLY_IGNORANT_ATTRIBUTES)
```

Comment:
Only one caller and it negates the result.  Invert this and the one usage is easier to read.

:question: